tests/test_flatten.py

```diff
@@ -31,3 +37,21 @@ def test_flatten_dict():
             "contextInfo_context2": "info2",
         }
     ]
+
+
+# Write a test that passes a pandas.DataFrame as data to the existing flatten function
+def test_flatten_gpd():
+    flattened_pd = flatten(
+        PD_DATA, nested_columns=["contextInfo", "neighborhood"]
+    )
+    assert flattened_pd.equal(
```

Comment:
Not sure here, but I understand that to validate if tow DataFrames are equals, we should use [`pd1.equal(pd2)`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.equals.html)

Reply:
BUT the [test is failing](https://github.com/FelipeSBarros/crossfire/actions/runs/8010990526/job/21883232212?pr=103#step:6:46) in the [data validation](https://github.com/FelipeSBarros/crossfire/blob/make_flatten_gdf/crossfire/__init__.py#L58):
```python
def flatten(data, nested_columns=None):
    nested_columns = set(nested_columns or NESTED_COLUMNS)
    if not nested_columns.issubset(NESTED_COLUMNS):
        raise NestedColumnError(nested_columns)
    if not data:
        return data
```
This is because, when dealing with `DataFrame` we need to use different approaches, like `df1.notnull()`, in instance to validate if a DataFram is passed bu it is None...
Thought about passing this validation to underneth of data type validation, and write a new validation to "identify" `DataFrame`s.
Something like:
```python
def flatten(data, nested_columns=None):
    nested_columns = set(nested_columns or NESTED_COLUMNS)
    if not nested_columns.issubset(NESTED_COLUMNS):
        raise NestedColumnError(nested_columns)
    if isinstance(data, list):
        if not data:
            return data
        keys = set(data[0].keys()) & nested_columns
    if isinstance(data, DataFrame):
        if data.isnull():
            return data

```

Is it a messy approach?

Reply:
If the tests are failing you are still on step 2. Asking _is it too messy_ is step 3.

By now focus on making the test pass.